network/topology/randomizedTopology.go

```diff
@@ -0,0 +1,165 @@
+package topology
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/crypto/random"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+// RandomizedTopology generates a random topology per channel.
+// By random topology we mean a node is connected to any other co-channel nodes with some
+// edge probability.
+type RandomizedTopology struct {
+	me      flow.Identifier             // used to keep identifier of the node
+	state   protocol.State              // used to keep a read only protocol state
+	subMngr network.SubscriptionManager // used to keep track topics the node subscribed to
+	chance  uint64                      // used to translate connectedness probability into a number in [0, 100]
```

Comment:
@yhassanzadeh13 what is the best way to interpret this? Is this the probability that two nodes would be connected?
Can I say that if node 1 has 10% probability of being down and node 2 has 20% probability of being down then `change` should be 30?
I am trying to relate it to something we know heuristically - node availability